containerd-shim-spin/src/engine.rs

```diff
@@ -62,6 +62,18 @@ impl Engine for SpinEngine {
 
     fn run_wasi(&self, ctx: &impl RuntimeContext, stdio: Stdio) -> Result<i32> {
         stdio.redirect()?;
+
+        // Set the container environment variables which will be collected by Spin's
+        // [environment variable provider].
+        // TODO: This is a temporary solution to allow Spin to collect the container environment variables.
+        // We should later look into other variable providers to collect container variables.
+        //
+        // [environment variable provider]: https://github.com/fermyon/spin/blob/v3.0.0/crates/variables/src/env.rs
```

Comment:
Right now, we use container environment variables to both configure application variables (as you mention) and configure the spin runtime (for OTEL, HTTP listen address, and other trigger configuration). Could you update the comment to also include the latter context as well?

Reply:
Yeah, I am aware of that. I will add the context to the this comment. 

Btw, I think we can extract a pre-defined list of Spin runtime variables, such as HTTP listen address, find and remove them from `ctx.env()` list. Then we don't need to populate Spin runtime variables to application variables, right? 

I forgot the initial reason for exposing both spin runtime and application variables to applications.

Reply:
We actually need to put many of them in the environment for the spin runtime. Some are not surfaces as CliArgs rather the triggers look in the env at execution time, for example the SQS trigger